src/lib/models/base/image-base-model.ts

```diff
@@ -24,4 +25,16 @@ export class ImageBaseModel<
     });
     return response.data;
   }
+
+  async generateWithFormData(body: RequestType): Promise<ResponseType> {
```

Comment:
The way I was imagining it, we should pick the right body (i.e json or multi-part) in code (maybe hardcoded for some request types, maybe dynamic based on provided arguments (i.e if the user provides base64, send as json, otherwise send as multipart).

But not the user having to decide how to send it (because in the end of the day this is an implementation detail).

I still don't see what possible options are supported (other than fs.createReadStream being able to work on them).

Reply:
Also, for some requests, there are other arguments (in addition to the image), so those have to also be sent as another `part` in the multipart body (and encoded properly, there it gets a bit dicey). E2E testing should help here :)